src/Servers/Kestrel/Core/src/BadHttpRequestException.cs

```diff
@@ -11,6 +12,7 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core
 {
+    [Obsolete("Moved to Microsoft.AspNetCore.Http.BadHttpRequestException")]
```

Comment:
As @Tratcher said, we should make this and the IIS BadHttpRequestException derive from Microsoft.AspNetCore.Http.BadHttpRequestException and we should continue throwing the more-derived types for max back-compat. Sorry I forgot to mention this earlier.

I think we can look at making the Reason enum public in a separate PR. I'm not sure the entire enum should be public. A lot of the Reasons are internal to Kestrel and would never be set on an exception that could be caught by the application.